src/Sentry.Unity/Il2CppEventProcessor.cs

```diff
@@ -56,47 +54,163 @@ public void Process(Exception incomingException, SentryEvent sentryEvent)
 
                 var nativeStackTrace = GetNativeStackTrace(exception);
 
-                if (!didAddImage)
-                {
-                    var codeFile = nativeStackTrace.ImageName;
-                    // NOTE: il2cpp in some circumstances does not return a correct `ImageName`.
-                    // A null/missing `CodeFile` however would lead to a processing error in sentry.
-                    // Since the code file is not strictly necessary for processing, we just fall back to
-                    // a sentinel value here.
-                    if (String.IsNullOrEmpty(codeFile))
-                    {
-                        codeFile = "GameAssembly.fallback";
-                    }
-                    debugImages.Add(new DebugImage
-                    {
-                        Type = _sentryUnityInfo.Platform,
+                _options.DiagnosticLogger?.LogDebug("NativeStackTrace Image: '{0}' (UUID: {1})", nativeStackTrace.ImageName, nativeStackTrace.ImageUuid);
 
-                        CodeFile = codeFile,
-                        DebugId = nativeStackTrace.ImageUuid,
-                        ImageAddress = $"0x{imageAddress:X8}",
-                    });
-
-                    didAddImage = true;
-                }
+                // Unity by definition only builds a single library which we add once to our list of debug images.
+                // We use this when we encounter stack frames with relative addresses.
+                // We want to use an address that is definitely outside of any address range used by real libraries.
+                // Canonical addresses on x64 leave a gap in the middle of the address space, which is unused.
+                // This is a range of addresses that we should be able to safely use.
+                // See https://en.wikipedia.org/wiki/X86-64#Virtual_address_space_details
+                var mainLibOffset = (ulong)1 << 63;
+                DebugImage? mainLibImage = null;
 
+                // TODO do we really want to continue if these two don't match?
+                //      Wouldn't it cause invalid frame info?
```

Comment:
good point, it probably would, I suspect it would only be on the bottom of the stack though.